qcodes/instrument_drivers/Keysight/SD_common/SD_AWG.py

```diff
@@ -691,6 +705,8 @@ def trigger_channels(self, channels: List[int]):
             channels: List of chnanels to trigger
         """
         # AWG channel mask, where LSB is for ch0, bit 1 is for ch1 etc.
+        if not self.zero_based_channels:
+            channels = [channel - 1 for channel in channels]
```

Comment:
Looks a bit redundant to have the same code lines repeated in multiple functions...

Reply:
For this PR I am okay with the code redundancy.